scripts/host-backup-restore/host-backup

```diff
@@ -23,11 +29,19 @@ tar --exclude 'tmp/*' --exclude 'tmp/.*' --exclude 'var/crash/*' --exclude 'var/
     --exclude 'var/log/audit*' \
     --exclude 'var/xsconfig/*' --exclude 'var/xsconfig/.*' \
     --exclude 'var/xapi/[0-9a-f]*-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*-[0-9a-f]*' \
+    --exclude-from ${EXCLUDE_FILE} --exclude ${EXCLUDE_FILE} \
     --sparse --preserve-permissions --to-stdout --gzip --one-file-system -C / -c . 
 
-# Remove the backup of the pool-database.  We only need it to be present in the tar file.
+TAR_EXIT_STATUS=$?
+
+# Remove the backup of the pool-database and the exclude list.
+# We only need it to be present in the tar file.
 # During restore the tar file is extracted over the backup partition which is then made
 # bootable.  Following a reboot the user must manually call "xe pool-restore-database" on 
 # the pool-database backup file.
 
-rm -f ${POOL_DB_BACKUP}
+rm -f ${POOL_DB_BACKUP} ${EXCLUDE_FILE}
+
+# if tar failed we need to know
+exit $TAR_EXIT_STATUS
```

Comment:
Instead of doing this, you should run `setopt -e` early in the script (or run it with `bash -e`) so that bash exits immediately if any command returns a non-zero error code.